packages/block-editor/src/components/media-upload/index.native.js

```diff
@@ -23,16 +23,27 @@ export const MEDIA_UPLOAD_BOTTOM_SHEET_VALUE_WORD_PRESS_LIBRARY = 'wordpress_med
 
 export const OPTION_TAKE_VIDEO = __( 'Take a Video' );
 export const OPTION_TAKE_PHOTO = __( 'Take a Photo' );
+export const OPTION_TAKE_PHOTO_OR_VIDEO = __( 'Take a Photo or Video' );
 
 export class MediaUpload extends React.Component {
+	constructor( props ) {
+		super( props );
+		this.onPickerPresent = this.onPickerPresent.bind( this );
+		this.onPickerChange = this.onPickerChange.bind( this );
+		this.onPickerSelect = this.onPickerSelect.bind( this );
+	}
 	getTakeMediaLabel() {
-		const { mediaType } = this.props;
+		const { allowedTypes = [] } = this.props;
+
+		const isOneType = allowedTypes.length === 1;
+		const isImage = isOneType && allowedTypes.includes( MEDIA_TYPE_IMAGE );
+		const isVideo = isOneType && allowedTypes.includes( MEDIA_TYPE_VIDEO );
 
-		if ( mediaType === MEDIA_TYPE_IMAGE ) {
+		if ( isImage ) {
 			return OPTION_TAKE_PHOTO;
-		} else if ( mediaType === MEDIA_TYPE_VIDEO ) {
+		} else if ( isVideo ) {
 			return OPTION_TAKE_VIDEO;
-		}
+		} return OPTION_TAKE_PHOTO_OR_VIDEO;
 	}
```

Comment:
Generally I've updated `media-upload` to check if `allowedTypes` contains more than one type, however, I think that on mobile doesn't exist a block which is going to support both types at the same time. 

For Picker (bottom sheet) I had to add a default label for that case `OPTION_TAKE_PHOTO_OR_VIDEO`, but I'm missing a proper icon in function `getChooseFromDeviceIcon`.

Please, let me know if everything is correct.

Reply:
I think we can default to image icon 'format-image' for OPTION_TAKE_PHOTO_OR_VIDEO.

<img width="319" alt="Screen Shot 2019-08-23 at 18 45 03" src="https://user-images.githubusercontent.com/5032900/63605242-325ade80-c5d6-11e9-8180-ff031cfe49b1.png">

Reply:
Done!

Reply:
> I think that on mobile doesn't exist a block which is going to support both types at the same time.

It doesn't today, but we hope to have the media-text block available soon which would allow both images and video

Reply:
I think [admin-media](https://developer.wordpress.org/resource/dashicons/#admin-media) is the more generic option, but I'd ask @iamthomasbishop

Reply:
In that place, we are using `Dashicon` and currently, we are supporting only video or image, but there is a pending PR which is adding multiple types so I would like to prepare a functionality for that. The only missing item is the `icon` for the case where both types are supported.

Reply:
So right now there is a `format-image`  as @pinarol 	suggested, but if you  have better idea, let me know.